src/style_manager/index.ts

```diff
@@ -597,27 +597,65 @@ export default class StyleManager extends ItemManagerModule<
       const cmp = target.toHTML ? target : target.getComponent();
       const optsSel = { array: true } as const;
       let cmpRules: CssRule[] = [];
+      let tagNameRules: CssRule[] = [];
+      let invisibleRules: CssRule[] = [];
       let otherRules: CssRule[] = [];
       let rules: CssRule[] = [];
 
       const rulesBySelectors = (values: string[]) => {
-        return !values.length
-          ? []
-          : cssC.getRules().filter(rule => {
-              const rSels = rule.getSelectors().map(s => s.getFullName());
-              return !!rSels.length && rSels.every(rSel => values.indexOf(rSel) >= 0);
+        return cssC.getRules().filter(rule => {
+          const rSels = rule.getSelectors().map(s => s.getFullName());
+
+          // rSels is equal to 0 when rule selectors contain tagName like : p {}, .logo path {}, ul li {}
+          if (rSels.length === 0) {
+            return false;
+          }
+
+          return rSels.every(rSel => values.indexOf(rSel) >= 0);
+        });
+      };
+
+      const rulesByTagName = (tagName: string) => {
+        return cssC.getRules().filter(rule => {
+          const ruleTags = rule
+            .selectorsToString()
+            .split(' ')
+            .filter(item => {
+              return item.startsWith('.') === false && item.startsWith('#') === false;
             });
+
+          if (ruleTags.length === 0) {
+            return false;
+          }
+
+          const lastTags = ruleTags[ruleTags.length - 1];
+
+          return lastTags === tagName;
+        });
       };
 
       // Componente related rule
       if (cmp) {
         cmpRules = cssC.getRules(`#${cmp.getId()}`);
+        tagNameRules = rulesByTagName(cmp.get('tagName'));
         otherRules = sel ? rulesBySelectors(sel.getSelectors().getFullName(optsSel)) : [];
-        rules = otherRules.concat(cmpRules);
+        rules = otherRules.concat(tagNameRules).concat(cmpRules);
       } else {
         cmpRules = sel ? cssC.getRules(`#${sel.getId()}`) : [];
+        tagNameRules = rulesByTagName(sel?.get('tagName') || '');
+        // Get rules set on active and visible selectors
         otherRules = rulesBySelectors(target.getSelectors().getFullName(optsSel));
-        rules = cmpRules.concat(otherRules);
+        // Get rules set on invisible selectors like private one
+        const allCmpClasses = sel?.getSelectors().getFullName(optsSel) || [];
+        const invisibleSel = allCmpClasses.filter(
+          (item: string) =>
+            target
+              .getSelectors()
+              .getFullName(optsSel)
+              .findIndex(sel => sel === item) === -1
+        );
+        invisibleRules = rulesBySelectors(invisibleSel);
```

Comment:
`rulesBySelectors` is already executed above, it would make sense to combine CssRule selectors + Component ones and pass them to `rulesBySelectors` together

Reply:
Of course, I update code to call `rulesBySelectors` only once!